### ISSUE TITLE
exceed 2^31-1 bytes in methylDB object during conversion to non-DB variant

### ISSUE THREAD

**Issue by alexg9010** [al2na/methylKit#141]
People are running into an issue when trying to convert a big methlDB object into the in-memory variant. 
This Code caused the explicit error:
```
> objDB=as(methDB,"methylBase")

Error in paste(tabixRes[[1]], collapse = "\n") : 
  result would exceed 2^31-1 bytes
```

I was able to pinpoint the problematic line (https://github.com/al2na/methylKit/blob/master/R/tabix.functions.R#L317) which is called when we are using a select or subsetting (`[`) call. 

```
select/`[` --> headTabix --> getTabixByChunk --> tabix2dt --> fread(paste(tabixRes[[1]],collapse="\n"),"\n" )
```

, where tabixRes is a list, with one element per region. Each element of the list is a character vector representing records in the region. 
The error occurs because size of the pasted string exceeds the length limit of strings (
https://stackoverflow.com/questions/53120436/error-in-pastev-collapse-n-result-would-exceed-231-1-bytes ) .

**Comment by al2na** [comment]
any idea how to fix it?

**Comment by alexg9010** [comment]
My idea is to replace `headTabix`/ `getTabixByChunk` with `applyTabixByChunk` in all functions that fetch the whole methylDB into memory, such that this will be done by chunks (of 1e6 lines). This way we should not reach the string length anymore.

**Comment by al2na** [comment]
maybe we can look at what these guys are doing:
https://www.rdocumentation.org/packages/seqminer/versions/6.7/topics/tabix.read.table

**Comment by al2na** [comment]
you can also look at the example in rsamtools, I don't know how fast it
would be, they use map instead of reading everything at once.

res <- scanTabix(tbx)
dff <- Map(function(elt) { read.csv(textConnection(elt), sep="\t",
header=FALSE) }, res)

**Comment by katwre** [comment]
yes, I sometimes use data.table::fread to read even really big tabix datasets (1.7G) and it's quite fast, e.g.:
```
fread('zcat ~/methylBase_meth.deT.nodups.txt.bgz')
```

**Comment by alexg9010** [comment]
yes, I would like to stay with fread too, and we are already using it extensively. 
tabix2dt and alikes where supposed to only read chunks and not the whole files, so I think we need to limit their use to those cases.

**Comment by alexg9010** [comment]
should be fixed in aa6a0d122345ed31e73f226f5733da596e96ad51
